integration-tests/src/test/java/oracle/weblogic/kubernetes/ItLifecycleSampleScripts.java

```diff
@@ -0,0 +1,432 @@
+  public static final String CLUSTER_LIFECYCLE = "Cluster";
+  public static final String DOMAIN = "DOMAIN";
+  public static final String STOP_SERVER_SCRIPT = "stopServer.sh";
+  public static final String START_SERVER_SCRIPT = "startServer.sh";
+  public static final String STOP_CLUSTER_SCRIPT = "stopCluster.sh";
+  public static final String START_CLUSTER_SCRIPT = "startCluster.sh";
+  public static final String STOP_DOMAIN_SCRIPT = "stopDomain.sh";
+  public static final String START_DOMAIN_SCRIPT = "startDomain.sh";
+  private static String domainNamespace = null;
+  private static final String domainName = "domain1";
+  private final int replicaCount = 2;
+  private final String clusterName = "cluster-1";
+  private final String adminServerName = "admin-server";
+  private final String managedServerNameBase = "managed-server";
+
+  private final String adminServerPodName = domainName + "-" + adminServerName;
+  private final String managedServerPodNamePrefix = domainName + "-" + managedServerNameBase;
+  private final Path samplePath = Paths.get(ITTESTS_DIR, "../kubernetes/samples");
+  private final Path domainLifecycleSamplePath = Paths.get(samplePath + "/scripts/domain-lifecycle");
+  private final Path tempSamplePath = Paths.get(WORK_DIR, "lifecycle-scripts-testing");
+  private final Path sampleBase =
+          Paths.get(tempSamplePath.toString(), "scripts/create-weblogic-domain/domain-home-on-pv");
+
+  // create standard, reusable retry/backoff policy
+  private static final ConditionFactory withStandardRetryPolicy
+      = with().pollDelay(2, SECONDS)
+          .and().with().pollInterval(10, SECONDS)
+          .atMost(10, MINUTES).await();
+
+  private static LoggingFacade logger = null;
+
+  /**
+   * Assigns unique namespaces for operator and domains and installs operator.
+   *
+   * @param namespaces injected by JUnit
+   */
+  @BeforeAll
+  public void initAll(@Namespaces(2) List<String> namespaces) {
+    String opNamespace = namespaces.get(0);
+
+    logger = getLogger();
+
+    logger.info("Assign a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace is null");
+    logger.info("Assign a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace is null");
+    domainNamespace = namespaces.get(1);
+
+    // create pull secrets for WebLogic image when running in non Kind Kubernetes cluster
+    // this secret is used only for non-kind cluster
+    createSecretForBaseImages(domainNamespace);
+
+    // install operator and verify its running in ready state
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    //create and start WebLogic domain using domain-home-on-pv sample scripts
+    createDomain(sampleBase);
+  }
+
+  /**
+   * Test scripts for stopping and starting a managed server.
+   */
+  @Test
+  @DisplayName("Test server lifecycle samples scripts")
+  public void testServerLifecycleScripts() {
```

Comment:
You can add these tests into existing ItSamples.java. You can use the domain that's already created/started by the test in ItSamples to verify your scripts. That saves time for one domain creation/startup and all the tests for verifying the samples will be in one class.

Reply:
I'm not sure if adding these tests as it is in ItSamples.java will work since ItSamples creates 2 types of domains (using WLST and WDT) and domain lifecycle tests will get executed twice (once for each type of domain). That'll increase the test execution time. Also, coupling the two tests with different goals in the same class might make it hard to understand and maintain. Domain creation is a pre-requisite for lifecycle scripts testing and if saving time is a concern, maybe we can prioritize and run `ItSamples` or `ItLifecycleSampleScripts` test less frequently (e.g. weekly instead of nightly). Please let me know what you think. Thanks.

Reply:
The test you are going to add is not a parameterized test, it will be executed only once. Your test is going use the domain that is started by the existing test. I think the goal is same, test sample scripts.

Reply:
It seems ItSamples needs some refactoring. I tried adding above tests in ItSamples and got below error. Looks like domain is not created when stopServer.sh script got executed.

<10-22-2020 21:23:15> <INFO> <oracle.weblogic.kubernetes.actions.impl.primitive.Command execute> <Executing command sh /scratch/akedia/wls-k8s-files/weblogic-kubernetes-operator/integration-tests/../kubernetes/samples/scripts/domain-lifecycle/stopServer.sh -d domain1 -n ns-svss -s managed-server1 >
<10-22-2020 21:23:15> <SEVERE> <oracle.weblogic.kubernetes.actions.impl.primitive.Command execute> <The command execution failed because it returned non-zero exit value: ExecResult: exitValue = 1, stdout = , stderr = Error from server (NotFound): pods "domain1-managed-server1" not found.>

Reply:
Anil, It could be because of the order in which the tests are executed. Your lifecycle tests might have executed before the test method that creates the domain. You can use Order annotation to execute the tests in the way you want. Please see ItIntrospectVersion.java for example for test ordering.

Reply:
@sankarpn and I looked at it together. Currently domain creation and deletion method are not generic and it makes certain assumption about directory structure. For e.g. files created for domain1 are re-used for domain2 and domain1 directories are deleted before domain2 creation begins. Hence deletion of both domains towards the end of test fails. I can look into this tomorrow or as Pani suggested below, it can be done as part of another integration tests JIRA.

Reply:
Fixed. With Sankar's help, the test methods are merged in ItSamples and I have deleted ItLifecycleSampleScripts.java.